components/mediation-commons/src/main/java/org/wso2/carbon/mediation/commons/internal/MediationCommonsComponent.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.mediation.commons.internal;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.wso2.carbon.utils.ConfigurationContextService;
+
+/**
+ * Declarative service component Integrator component.
+ */
+@Component(
+        name = "org.wso2.carbon.mediation.commons.internal.MediationCommonsComponent",
+        immediate = true)
+public class MediationCommonsComponent {
```

Comment:
Let's investigate possibility of moving the service component outside this org.wso2.carbon.mediation.commons bundle or create/introduce new interface and add reference to it instead of directly depending on org.wso2.carbon.utils